sstable/block_property_obsolete.go

```diff
@@ -6,56 +6,69 @@ package sstable
 
 import "github.com/cockroachdb/errors"
 
+// obsoleteKeyBlockPropertyCollector is a block property collector used to
+// implement obsoleteKeyBlockPropertyFilter - a filter that excludes blocks
+// which contain only obsolete keys.
+//
+// For an explanation of obsolete keys, see the comment for TableFormatPebblev4
+// which explains obsolete keys.
 type obsoleteKeyBlockPropertyCollector struct {
 	blockIsNonObsolete bool
 	indexIsNonObsolete bool
 	tableIsNonObsolete bool
 }
 
-func encodeNonObsolete(isNonObsolete bool, buf []byte) []byte {
-	if isNonObsolete {
-		return buf
-	}
-	return append(buf, 't')
-}
+var _ BlockPropertyCollector = (*obsoleteKeyBlockPropertyCollector)(nil)
 
+// Name is part of the BlockPropertyCollector interface.
 func (o *obsoleteKeyBlockPropertyCollector) Name() string {
 	return "obsolete-key"
 }
 
+// Add is part of the BlockPropertyCollector interface.
 func (o *obsoleteKeyBlockPropertyCollector) Add(key InternalKey, value []byte) error {
 	// Ignore.
 	return nil
 }
 
+// AddPoint is an out-of-band method that is specific to this collector.
 func (o *obsoleteKeyBlockPropertyCollector) AddPoint(isObsolete bool) {
 	o.blockIsNonObsolete = o.blockIsNonObsolete || !isObsolete
 }
 
+// FinishDataBlock is part of the BlockPropertyCollector interface.
 func (o *obsoleteKeyBlockPropertyCollector) FinishDataBlock(buf []byte) ([]byte, error) {
 	o.tableIsNonObsolete = o.tableIsNonObsolete || o.blockIsNonObsolete
-	return encodeNonObsolete(o.blockIsNonObsolete, buf), nil
+	return obsoleteKeyBlockPropertyEncode(!o.blockIsNonObsolete, buf), nil
 }
 
+// AddPrevDataBlockToIndexBlock is part of the BlockPropertyCollector interface.
 func (o *obsoleteKeyBlockPropertyCollector) AddPrevDataBlockToIndexBlock() {
 	o.indexIsNonObsolete = o.indexIsNonObsolete || o.blockIsNonObsolete
 	o.blockIsNonObsolete = false
 }
 
+// FinishIndexBlock is part of the BlockPropertyCollector interface.
 func (o *obsoleteKeyBlockPropertyCollector) FinishIndexBlock(buf []byte) ([]byte, error) {
-	indexIsNonObsolete := o.indexIsNonObsolete
+	isObsolete := true
+	if o.indexIsNonObsolete {
```

Comment:
nit: could you pass `!o.indexIsNonObsolete` into `obsoleteKeyBlockPropertyEncode` ?